src/ast/codegen_llvm.cpp

```diff
@@ -557,8 +557,16 @@ void CodegenLLVM::visit(Call &call)
       arg.offset = struct_layout->getElementOffset(i+1); // +1 for the printf_id field
     }
 
-    AllocaInst *printf_args = b_.CreateAllocaBPF(printf_struct, "printf_args");
-    b_.CreateMemSet(printf_args, b_.getInt8(0), struct_size, 1);
+    Value *printf_args = b_.CreateGetPrintfMap(printf_struct);
+
+    Function *parent = b_.GetInsertBlock()->getParent();
```

Comment:
is this solely an optimization to skip the probe_reads and perf_event_output in the case of a zero-arg printf?

Reply:
Yes, this handles a failure to acquire the map entry.